File: 206-reverse-linked-list/reverse-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function reverseList(head: ListNode | null): ListNode | null {
    if (head === null) return null

    let current: ListNode = head
    let next: ListNode = null;
    let prev: ListNode = null;

    while (current !== null) {
        next = current.next
        current.next = prev

        prev = current
        current = next
    }

    return prev
    
};

File: 206-reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head

        curr = head
        prev = None
        _next = None

        while curr:
            _next = curr.next
            curr.next = prev
            prev = curr
            curr = _next

        return prev